spec/requests/devise/registrations/user_signup_spec.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'external user not created' do
+  it { expect { request }.to change(User, :count).by(0) }
+  it { expect { request }.to change(ExternalUser, :count).by(0) }
+  it { expect { request }.to change(Provider, :count).by(0) }
+end
+
+RSpec.shared_examples 'external user created' do
+  it { expect { request }.to change(User, :count).by(1) }
+  it { expect { request }.to change(ExternalUser, :count).by(1) }
+  it { expect { request }.to change(Provider, :count).by(1) }
+end
+
+RSpec.describe 'User sign up', type: :request do
+  include Capybara::RSpecMatchers
+
+  describe 'GET #new' do
+    subject(:request) { get new_user_registration_path }
+
+    context 'when NOT on api-sandbox' do
+      around do |example|
+        with_env('production') do
+          example.run
+        end
+      end
+
+      it {
+        request
```

Comment:
Can this `request` (and likewise below on lines 42 and 47) be moved into the `around` (or `before`) block?

Reply:
i tried this and it was not working in the around block. using a before block would not work, i believe, as the request must be made after the env var is set. A before and after block might work but would meaning changing the ENV var helper possibly.